gensim/models/sent2vec.py

```diff
@@ -0,0 +1,827 @@
+        for i in range(len(counts)):
+            c = counts[i] ** 0.5
+            for j in range(int(c * self.negative_table_size / z) + 1):
+                self.negatives.append(i)
+        self.random.shuffle(self.negatives)
+        self.negatives = np.array(self.negatives)
+
+    def get_negative(self, target):
+        """
+        Get a negative from the list of negatives for caluculating nagtive sampling loss.
+        Parameter
+        ---------
+        target : int
+            Target word id.
+        Returns
+        -------
+        int
+            Word id of negative sample.
+        """
+
+        while True:
+            negative = self.negatives[self.negpos]
+            self.negpos = (self.negpos + 1) % len(self.negatives)
+            if target != negative:
+                break
+        return negative
+
+    def _do_train_job(self, sentences, lr, hidden, grad):
+
+        if FAST_VERSION == -1:
+            local_token_count = 0
+            nexamples = 0
+            loss = 0.0
+            for sentence in sentences:
+                ntokens_temp, hashes, words = self.dict.get_line(sentence)
+                local_token_count += ntokens_temp
+                if len(words) > 1:
+                    for i in range(len(words)):
+                        if self.random.uniform(0, 1) > self.dict.pdiscard[words[i]]:
+                            continue
+                        nexamples += 1
+                        context = list(words)
+                        context[i] = 0
+                        context = self.dict.add_ngrams_train(context=context, n=self.word_ngrams, k=self.dropoutk)
+                        loss += self.update(input_=context, target=words[i], lr=lr)
+            return local_token_count, nexamples, loss
+        else:
+            local_token_count, nexamples, loss = _do_train_job_fast(self, sentences, lr, hidden, grad)
+            return local_token_count, nexamples, loss
+
+    def update(self, input_, target, lr):
+        """
+        Update model's neural weights for given context, target word and learning rate.
+        Parameters
+        ----------
+        input_ : list
+            List of word ids of context words.
+        target : int
+            Word id of target word.
+        lr : float
+            Current Learning rate.
+        """
+
+        assert(target >= 0)
+        assert(target < self.dict.size)
+        if len(input_) == 0:
+            return
+        self.hidden = np.zeros(self.vector_size)
+        for i in input_:
+            self.hidden += self.wi[i]
+        self.hidden *= (1.0 / len(input_))
+        loss = self.negative_sampling(target, lr)
+        self.grad *= (1.0 / len(input_))
+        for i in input_:
+            self.wi[i] += self.grad
+        return loss
+
+    def build_vocab(self, sentences):
+        logger.info("Creating dictionary...")
+        self.dict = ModelDictionary(t=self.t, bucket=self.bucket, maxn=self.maxn, minn=self.minn)
```

Comment:
`max_vocab_size` never set. Pass `self.max_vocab_size` to `ModelDictionary` here.

Reply:
I'm using default `max_vocab_size` as specified in the `__init__()` function of `ModelDictionary`

Reply:
Shouldn't this be a parameter that the user can set (look at `Word2Vec`/`FastText`)? I can see that you have mentioned  this parameter's description in `Sent2Vec.__init__()` docstring but is missing from the method.

Reply:
Oh yes. My bad. I've fixed it in latest commit.